### PR TITLE
Withdraw flow

### PR DESCRIPTION
**Pull request by jesseyuen**
This PR includes the withdrawal flow which I imagine we'll keep tweaking this as we know more.

Looking forward to your feedback!

### CODE REVIEW

File: app/routes.js [hunk 1]
```diff
@@ -596,6 +596,64 @@ router.post('/record/:uuid/reinstate', (req, res) => {
...
+//     delete newDegree.baseGrade
+//     delete newDegree.otherGrade
+//   }
+//   else {
+//     newDegree.grade = newDegree.baseGrade
+//     delete newDegree.baseGrade
+//     delete newDegree.otherGrade
+//   }
+// }
+
```
Comment (edwardhorsford) [inline app/routes.js]: Can these be deleted?

File: app/views/_includes/forms/withdraw.html [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        }
+      ]
+  }) }}
+{% endset %}
+
+{{ govukRadios({
+  fieldset: {
+    legend: {
+      text: "When did the trainee withdraw?",
+      classes: "govuk-fieldset__legend--s govuk-!-margin-top-4 govuk-!-margin-bottom-4"
```
Comment (edwardhorsford) [inline app/views/_includes/forms/withdraw.html]: Question: are these margin overrides intentional?

File: app/views/_includes/summary-cards/student-record.html [hunk 3]
```diff
@@ -96,15 +95,14 @@
...
-      classes: "govuk-tag--yellow govuk-!-margin-bottom-2"
-    })}} <br>
-    Deferral date: {{record.deferredDate | govukDate}}
-{% endset %}
-
 {% if (record.status == "Deferred") %}
+  {% set deferredContent %}
+      {{govukTag({
+        text: "Deferred",
+        classes: "govuk-tag--yellow govuk-!-margin-bottom-2"
```
Comment (edwardhorsford) [inline app/views/_includes/summary-cards/student-record.html]:
```suggestion
        classes: record.status | getStatusClass + " govuk-!-margin-bottom-2"
```

File: app/views/_includes/summary-cards/student-record.html [hunk 4]
```diff
@@ -116,6 +114,25 @@
   }) %}
 {% endif %}
 
+{% if (record.status == "Withdrawn") %}
+  {% set withdrawalContent %}
+      {{govukTag({
+        text: "Withdrawn",
+        classes: "govuk-tag--red govuk-!-margin-bottom-2"
```
Comment (edwardhorsford) [inline app/views/_includes/summary-cards/student-record.html]:
```suggestion
        classes: record.status | getStatusClass + " govuk-!-margin-bottom-2"
```

File: scripts/generate-records.js [hunk 5]
```diff
@@ -59,6 +59,13 @@ const generateFakeApplication = (params = {}) => {
       moment(submittedDate),
       moment(updatedDate))
   }
+  // If the status is Withdrawn 
+  if (status === 'Withdrawn') {    
+    // Make sure deferral date is between submitted and updated date
+    withdrawalDate = params.withdrawalDate || faker.date.between(
+      moment(submittedDate),
+      moment(updatedDate))
+  }
```
Comment (edwardhorsford) [inline scripts/generate-records.js]: I think you could just make this equal to the updated date.

**Review by edwardhorsford — CHANGES_REQUESTED** [review]

File: app/views/_includes/forms/withdraw.html (hunk 2 above)
Comment (jesseyuen) [reply app/views/_includes/forms/withdraw.html]: They were... for now. Looking at these, it's a flow on from a duplicated markup (H1) issue. Sorting it.

**Review by edwardhorsford — APPROVED** [review]

**Pull request merged**